mitmproxy/tools/console/tabs.py

```diff
@@ -50,7 +50,7 @@ def show(self):
         headers = []
         for i in range(len(self.tabs)):
             txt = self.tabs[i][0]()
-            if i == self.tab_offset:
+            if i == self.tab_offset or (len(self.tabs) <= self.tab_offset and i == 0):
```

Comment:
Style tip for this type of code: At the moment it's really hard to parse what is actually happening here. You can alleviate this by introducing a variable with a speaking name:
```python
is_focused = (
    i == self.tab_offset
    or (i == 0 and self.tab_offset >= len(self.tabs)):
)
if is_focused:
    # ...
```

I like that you don't modify `.tab_offset` if we have less tabs: That would be annoying when tabbing HTTP flow -> TCP flow -> HTTP flow as it would reset the offset. We could however simplify the implementation a bit and use `self.tab_offset % len(self.tabs)` everywhere.

Reply:
Thanks for the tip